### PR TITLE
Bump images to Golang 1.22

### PR DESCRIPTION
**Pull request by skonto**
**What this PR does / why we need it**:

-  Similar to https://github.com/openshift-knative/serving/pull/687
- See https://github.com/openshift-knative/serving/pull/749#issuecomment-2191574824
**Which issue(s) this PR fixes**:

JIRA:

**Does this PR needs for other branches**:
NONE

**Does this PR (patch) needs to update/drop in the future?**:

JIRA: NONE

### CODE REVIEW

File: openshift/ci-operator/Dockerfile.in [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Do not edit! This file was generated via Makefile
-FROM registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.21-openshift-4.16 as builder
+FROM registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.22-openshift-4.17 as builder
```
Comment (dsimansk) [inline openshift/ci-operator/Dockerfile.in:2]: Is there no RHEL 8 variant with 1.22? Because productized images will be still RHEL 8 based. Otherwise fine

Comment (skonto) [reply openshift/ci-operator/Dockerfile.in:2]: There is `registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.22-openshift-4.17` I can change to that.

Comment (skonto) [reply openshift/ci-operator/Dockerfile.in]: done

**Pull request merged**